src/main/java/redis/clients/jedis/BinaryJedisCluster.java

```diff
@@ -0,0 +1,1261 @@
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.zremrangeByScore(key, start, end);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Long zremrangeByScore(final byte[] key, final byte[] start, final byte[] end) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.zremrangeByScore(key, start, end);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Long linsert(final byte[] key, final Client.LIST_POSITION where, final byte[] pivot,
+      final byte[] value) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.linsert(key, where, pivot, value);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Long lpushx(final byte[] key, final byte[]... arg) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.lpushx(key, arg);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Long rpushx(final byte[] key, final byte[]... arg) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.rpushx(key, arg);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public List<byte[]> blpop(final byte[] arg) {
+    return new JedisClusterCommand<List<byte[]>>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public List<byte[]> execute(Jedis connection) {
+        return connection.blpop(arg);
+      }
+    }.runBinary(arg);
+  }
+
+  @Override
+  public List<byte[]> brpop(final byte[] arg) {
+    return new JedisClusterCommand<List<byte[]>>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public List<byte[]> execute(Jedis connection) {
+        return connection.brpop(arg);
+      }
+    }.runBinary(arg);
+  }
+
+  @Override
+  public Long del(final byte[] key) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.del(key);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public byte[] echo(final byte[] arg) {
+    // note that it'll be run from arbitary node
+    return new JedisClusterCommand<byte[]>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public byte[] execute(Jedis connection) {
+        return connection.echo(arg);
+      }
+    }.runBinary(arg);
+  }
+
+  @Override
+  public Long move(final byte[] key, final int dbIndex) {
```

Comment:
`move` is not allowed in cluster mode since there's only db at index `0` available. Leads to `(error) ERR MOVE is not allowed in cluster mode`